test/test_other.py

```diff
@@ -1307,6 +1307,52 @@ def test_export_all(self):
     self.emcc('lib.c', ['-Oz', '-sEXPORT_ALL', '-sLINKABLE', '--pre-js', 'main.js'], output_filename='a.out.js')
     self.assertContained('libf1\nlibf2\n', self.run_js('a.out.js'))
 
+  def test_export_keepalive(self):
+    create_file('main.c', r'''
+      #include <emscripten.h>
+      EMSCRIPTEN_KEEPALIVE int libf1() { return 42; }
+    ''')
+
+    create_file('main.js', '''
+      var Module = {
+        onRuntimeInitialized: function() {
+          console.log(Module._libf1 ? Module._libf1() : 'unexported');
+        }
+      };
+    ''')
+
+    # By default, all kept alive functions should be exported.
+    self.emcc('main.c', ['--pre-js', 'main.js'], output_filename='test.js')
```

Comment:
We normally write this as just one line: `self.do_runf(`main.c`, '42\n', emcc_args=['--pre-js', 'main.js'])`

(sorry, there are a lot ways to do things in the test framework).